app/enterprise.go

```diff
@@ -136,8 +142,18 @@ func (s *Server) initEnterprise() {
 		s.Notification = notificationInterface(s.FakeApp())
 	}
 	if samlInterface != nil {
+		// default to original interface.
 		s.Saml = samlInterface(s.FakeApp())
 		s.AddConfigListener(func(_, cfg *model.Config) {
+			if samlInterfaceNew != nil {
```

Comment:
It seems like we should also have this logic outside the config listener to set the interface initially. I am guessing the config listener runs enough times that it's not a problem? Still seems fishy though. Maybe we aught to do it anyway.

Reply:
Agreed, maybe we just define the callback first, call it, and also register it as a listener?